src/constants.rs

```diff
@@ -59,1279 +59,157 @@
-        let time = match self {
-            Energy | Power | Hydrogen | Oxygen | Utrium | Lemergium | Keanium | Zynthium
-            | Catalyst | Ops => return None,
-            // these comments copied directly from JavaScript 'constants.js' file.
-            // OH: 20,
-            Hydroxide => 20,
-            // ZK: 5,
-            ZynthiumKeanite => 5,
-            // UL: 5,
-            UtriumLemergite => 5,
-            // G: 5,
-            Ghodium => 5,
-            // UH: 10,
-            UtriumHydride => 10,
-            // UH2O: 5,
-            UtriumAcid => 5,
-            // XUH2O: 60,
-            CatalyzedUtriumAcid => 60,
-            // UO: 10,
-            UtriumOxide => 10,
-            // UHO2: 5,
-            UtriumAlkalide => 5,
-            // XUHO2: 60,
-            CatalyzedUtriumAlkalide => 60,
-            // KH: 10,
-            KeaniumHydride => 10,
-            // KH2O: 5,
-            KeaniumAcid => 5,
-            // XKH2O: 60,
-            CatalyzedKeaniumAcid => 60,
-            // KO: 10,
-            KeaniumOxide => 10,
-            // KHO2: 5,
-            KeaniumAlkalide => 5,
-            // XKHO2: 60,
-            CatalyzedKeaniumAlkalide => 60,
-            // LH: 15,
-            LemergiumHydride => 15,
-            // LH2O: 10,
-            LemergiumAcid => 10,
-            // XLH2O: 65,
-            CatalyzedLemergiumAcid => 65,
-            // LO: 10,
-            LemergiumOxide => 10,
-            // LHO2: 5,
-            LemergiumAlkalide => 5,
-            // XLHO2: 60,
-            CatalyzedLemergiumAlkalide => 60,
-            // ZH: 20,
-            ZynthiumHydride => 20,
-            // ZH2O: 40,
-            ZynthiumAcid => 40,
-            // XZH2O: 160,
-            CatalyzedZynthiumAcid => 160,
-            // ZO: 10,
-            ZynthiumOxide => 10,
-            // ZHO2: 5,
-            ZynthiumAlkalide => 5,
-            // XZHO2: 60,
-            CatalyzedZynthiumAlkalide => 60,
-            // GH: 10,
-            GhodiumHydride => 10,
-            // GH2O: 15,
-            GhodiumAcid => 15,
-            // XGH2O: 80,
-            CatalyzedGhodiumAcid => 80,
-            // GO: 10,
-            GhodiumOxide => 10,
-            // GHO2: 30,
-            GhodiumAlkalide => 30,
-            // XGHO2: 150,
-            CatalyzedGhodiumAlkalide => 150,
-        };
-        Some(time)
-    }
-
-    /// Helper function for deserializing from a string rather than a fake
-    /// integer value.
-    pub fn deserialize_from_str<'de, D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
-        let s: Cow<'de, str> = Cow::deserialize(d)?;
-        Self::from_str(&s).map_err(|_| {
-            D::Error::invalid_value(
-                Unexpected::Str(&s),
-                &"a known constant string in RESOURCES_ALL",
-            )
-        })
-    }
+/// Re-export of all constants related to `Resource`s.
+pub mod resource {
+    pub use super::{numbers::ENERGY_DECAY, types::ResourceType};
```

Comment:
Might be worth adding `InterShardResource` here.

Reply:
I was thinking of `resource` mainly being about dropped resources / the `Resource` type. I think `IntershardResourceType`, while named similarly, is unrelated?

Reply:
Only problem is that it doesn't appear anywhere. Although, I guess it's still in `constants`...

Reply:
true, it doesn't really have another.. place. I mostly figured it'd be useful alone, like `Terrain` or `Direction`. Maybe all of those should have some place too, though? I'm not sure.